src/Db/PdoRepository.php

```diff
@@ -302,4 +309,175 @@ public function truncateWatches()
             $this->pdo->exec(sprintf('DELETE FROM %s', $this->tableWatches))
         );
     }
+
+    public function aggregate(array $options)
+    {
+        $query = sprintf('
+          SELECT
+            "id",
+            "request_ts",
+            "main_wt",
+            "main_ct",
+            "main_cpu",
+            "main_mu",
+            "main_pmu"
+          FROM %s
+          WHERE %s
+          ORDER BY %s %s',
+            $this->table,
+            $options['where']['conditions'],
+            $options['sort'],
+            $options['direction']
+        );
+        $stmt = $this->pdo->prepare($query);
+        $stmt->execute($options['where']['params']);
+
+        while ($row = $stmt->fetch(PDO::FETCH_ASSOC)) {
+            yield $row;
+        }
+    }
+
+    /**
+     * Convert request data keys into pdo query.
+     */
+    public function buildQuery(array $options): array
+    {
+        return [
+            'where' => $this->buildWhere($options['conditions']),
+            'sort' => $this->buildSort($options),
+            'direction' => $this->buildDirection($options),
+            'perPage' => $options['perPage'] ?? SearcherInterface::DEFAULT_PER_PAGE,
+        ];
+    }
+
+    /**
+     * build pdo where
+     *
+     * @param array $search
+     *
+     * @return array
+     */
+    public function buildWhere(array $search): array
+    {
+        $where = ['conditions' => '1=1', 'params' => []];
+
+        if (empty($search)) {
+            return $where;
+        }
+
+        if (!empty($search['limit_custom']) && $search['limit_custom'][0] === 'P') {
+            $search['limit'] = $search['limit_custom'];
+        }
+        $hasLimit = (!empty($search['limit']) && $search['limit'] != -1);
+
+        // simple_url equals match
+        if (isset($search['simple_url'])) {
+            $where['conditions'] .= ' and simple_url = :simple_url';
+            $where['params']['simple_url'] = $search['simple_url'];
+        }
+
+        if (!empty($search['date_start']) && !$hasLimit) {
+            $where['conditions'] .= ' and request_date >= :date_start';
+            $where['params']['date_start'] = $search['date_start'];
+        }
+
+        if (!empty($search['date_end']) && !$hasLimit) {
+            $where['conditions'] .= ' and request_date <= :date_end';
+            $where['params']['date_end'] = $search['date_end'];
+        }
+
+        if (!empty($search['request_start'])) {
+            $where['conditions'] .= ' and request_ts >= :request_start';
+            $where['params']['request_start'] = strtotime($search['request_start']);
+        }
+
+        if (!empty($search['request_end'])) {
+            $where['conditions'] .= ' and request_ts <= :request_end';
+            $where['params']['request_end'] = strtotime($search['request_end']);
+        }
+
+        // TODO need JSON support
+        if (!empty($search['remote_addr'])) {
+            $where['conditions'] .= ' and SERVER like :remote_addr';
```

Comment:
why LIKE?

Reply:
SERVER is json format, but not all databases support json operation